2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
+
+    # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
+    ip_and_port_regexp='[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+'
+    if addr="$(echo "$cmd_output" | grep -x "$ip_and_port_regexp")"; then
+      echo "$addr"
+      exit 0
+    fi
+
+    echo >&2 "Cannot get address of primary from $mongo_node node: $cmd_output"
```

Comment:
`cmd_output` will only contain the stdout of the mongo shell execution, or be empty if `true` was evaluated, right?

So I'm imagining in most cases it would be empty when we print this message, which is not very helpful, what do you think?